enterprise/utils.py

```diff
@@ -1091,3 +1091,15 @@ def delete_data_sharing_consent(course_id, customer_uuid, user_email):
     # Deleting the DCS cache
     consent_cache_key = get_cache_key(type='data_sharing_consent_needed', user_id=user.id, course_id=course_id)
     TieredCache.delete_all_tiers(consent_cache_key)
+
+
+def get_course_key_from_course_run_key(course_run_key):
+    """
+    Args:
+        course_run_key (str): The course run id for which we would like to know the parent course id.
+
+    Returns:
+        parent_course_key (str): The parent course id for the given course run.
+    """
+    split_key = course_run_key.split(':')[1].split('+')
+    return split_key[0] + '+' + split_key[1]
```

Comment:
FYI, I'm trying to find out if there's a more canonical way to do this.

Reply:
OK, course-discovery does something similar: https://github.com/edx/course-discovery/blob/3d928790bb9b71f12d0ed1e06af57970aee7c954/course_discovery/apps/course_metadata/utils.py#L348-L359 We should probably include the logic around `if '/' in course_run_key else ...` to handle the odd "old-style" course run keys.